Uplift/Views/Cells/FacilitiesDropdownCell.swift

```diff
@@ -0,0 +1,185 @@
+    private var collectionViewHeight: CGFloat = 0
+    private var dropdownView: DropdownView!
+    private var facility: Facility!
+    private var headerView: UIView!
+    private var height: CGFloat = 0
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        let collectionViewLayout = UICollectionViewFlowLayout()
+        collectionViewLayout.minimumInteritemSpacing = minimumInteritemSpacing
+
+        collectionView = UICollectionView(frame: .zero, collectionViewLayout: collectionViewLayout)
+        collectionView.delegate = self
+        collectionView.dataSource = self
+
+        setupDropdownView()
+        setupConstraints()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func configure(for facility: Facility) {
+        self.facility = facility
+        collectionView.reloadData()
+    }
+
+    func setupDropdownView() {
+        headerView = setupHeaderView(headerString: facility.name)
+
+        // TODO: register facility cell classes once they're finished
+
+        collectionViewHeight = 0 // TODO: once cell classes are finished, sum up their heights to get the total collection view height
+
+        dropdownView = DropdownView(delegate: self,
+                                    headerView: headerView,
+                                    headerViewHeight: headerViewHeight,
+                                    contentView: collectionView,
+                                    contentViewHeight: collectionViewHeight)
+    }
+
+    func setupHeaderView(headerString: String) -> UIView {
+        var headerImage: UIImage?
+        switch FacilityName(rawValue: headerString.uppercased()) {
+        case .equipment, .fitnessCenter:
+            headerImage = UIImage(named: ImageNames.equipment)
+        case .gymnasium:
+            headerImage = UIImage(named: ImageNames.basketball)
+        case .swimmingPool:
+            headerImage = UIImage(named: ImageNames.pool)
+        case .bowlingLanes:
+            headerImage = UIImage(named: ImageNames.bowling)
+        default:
+            headerImage = UIImage(named: ImageNames.misc)
+        }
+
+        headerImageView.image = headerImage
+        headerView.addSubview(headerImageView)
+
+        headerNameLabel.text = headerString
+        headerNameLabel.font = ._16MontserratRegular
+        headerNameLabel.textColor = .black
+        headerNameLabel.textAlignment = .left
+        headerView.addSubview(headerImageView)
+
+        let isFacilityOpen = getIsFacilityOpen()
+        headerOpenLabel.font = ._16MontserratRegular
+        headerOpenLabel.text = isFacilityOpen ? ClientStrings.CommonStrings.open : ClientStrings.CommonStrings.closed
+        headerOpenLabel.textColor = isFacilityOpen ? .accentOpen : .accentClosed
+        headerOpenLabel.textAlignment = .center
+        headerView.addSubview(headerOpenLabel)
+
+        return headerView
```

Comment:
why don't we just set `self.headerView = headerView` here instead of having to return a `UIView`?